app/views/imports/_form.html.erb

```diff
@@ -1,12 +1,35 @@
-<%= form_with(url: podcast_imports_path(import.podcast), model: [import.podcast, import.becomes(PodcastImport)], method: :post, html: {autocomplete: "off"}) do |form| %>
-  <div class="row my-4">
-    <div class="col-lg-8">
-      <%= render "form_main", imports: imports, podcast: podcast, form: form %>
+<%= form_with(model: import.becomes(PodcastImport), method: :post, html: {autocomplete: "off"}) do |form| %>
+  <%# hacky - but we need to cast back into the child class %>
+  <% form.object = import.becomes(import.class) %>
```

Comment:
what is the purpose of casting back into the child class?

Reply:
Okay, I had to try removing this to remember.

So we need to `.becomes(PodcastImport)` the parent class in the `form_with`... so that our field names are `podcast_import[url]` instead of `podcast_rss_import[url]`.

But then inside that, I needed to cast it back.  Otherwise calling something like `import.timings` would error - because that method doesn't exist on a `PodcastImport`.  Only on a `PodcastTimingsImport`.

I'll update the comment to be better.